src/redisearch.h

```diff
@@ -75,6 +75,8 @@ typedef enum {
   Document_HasPayload = 0x02,
   Document_HasSortVector = 0x04,
   Document_HasOffsetVector = 0x08,
+  Document_Expired = 0x10, // Document is expired but not yet marked as deleted. This is an optimization
```

Comment:
I'm not a fan of modifying a document with the read lock (even if it only serves the query). Could we put it in the search result?
Also, I'd rename it to "failed_to_open" or something for the option that in the future open_key might fail for additional reasons

Reply:
I agree mostly, but I moved it anyway for 2 reasons:

1. Removing the flags from the results making the struct smaller, which is significant when we buffer them.
2. Having 2 conditions on `r->dmd->flags` instead of two different flag objects actually “collapses” into a single assembly command, making it more efficient.

Regarding the rename, I agree. Will change